pkg/core/function.go

```diff
@@ -242,6 +248,7 @@ func (c *client) makeBuildSourceSpec(options CreateFunctionOptions) *build.Sourc
 }
 
 func (c *client) displayFunctionCreationProgress(serviceNamespace string, serviceName string, logWriter io.Writer, stopChan <-chan struct{}, errChan chan<- error) {
+	defer close(errChan)
```

Comment:
This avoids a deadlock in the `--local-path` case where `displayFunctionCreationProgress` was returning **without** closing `errChan`.